node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/provider/ComponentsProviderImpl.java

```diff
@@ -24,63 +25,52 @@
 import com.yahoo.vespa.hosted.node.admin.orchestrator.OrchestratorImpl;
 import com.yahoo.vespa.hosted.node.admin.util.ConfigServerHttpRequestExecutor;
 import com.yahoo.vespa.hosted.node.admin.util.Environment;
-import com.yahoo.vespa.hosted.node.admin.util.SecretAgentScheduleMaker;
 
-import java.io.IOException;
-import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.time.Clock;
-import java.util.Set;
+import java.time.Duration;
 import java.util.function.Function;
 
+import static com.yahoo.vespa.defaults.Defaults.getDefaults;
+
 /**
  * Set up node admin for production.
  *
  * @author dybis
  */
-public class ComponentsProviderImpl implements ComponentsProvider {
-    private static final ContainerName NODE_ADMIN_CONTAINER_NAME = new ContainerName("node-admin");
-
-    private final NodeAdminStateUpdater nodeAdminStateUpdater;
-    private final MetricReceiverWrapper metricReceiverWrapper;
-
-    private static final int NODE_AGENT_SCAN_INTERVAL_MILLIS = 30000;
+public class ComponentsProviderImpl extends AbstractComponent implements ComponentsProvider {
     private static final int WEB_SERVICE_PORT = getDefaults().vespaWebServicePort();
+    private static final Duration NODE_AGENT_SCAN_INTERVAL = Duration.ofSeconds(30);
```

Comment:
This is an internal detail that the node admin or agent should be responsible for.